analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -172,5 +172,83 @@ public void Execute_PersistSymbols_InsideBlock()
             static ISymbol LocalReferenceOperationSymbol(IOperationWrapperSonar operation) =>
                 ((ILocalReferenceOperation)operation.Instance).Local;
         }
+
+        [TestMethod]
+        public void ExitReached_SimpleFlow()
+        {
+            var exitCount = 0;
+            SETestContext.CreateCS("var a = true;", new ExitReachedTestCheck(state => CountExits(state, ref exitCount)));
+            Assert.AreEqual(1, exitCount);
+        }
+
+        [TestMethod]
+        public void ExitReached_MultipleBranches()
+        {
+            const string method = @"
+public int Get(bool a)
+{
+    if (a)
+        return 1;
+    else
+        return 2;
+}";
+            var exitCount = 0;
+            SETestContext.CreateCSMethod(method, new ExitReachedTestCheck(state => CountExits(state, ref exitCount)));
+            Assert.AreEqual(1, exitCount); // We don't support multiple branches yet.
+        }
+
+        [TestMethod]
+        public void ExitReached_Throw()
+        {
+            const string method = @"
+public int Get(bool a)
+{
+    if (a)
+        throw new System.NullReferenceException();
+    else
+        return 2;
+}";
```

Comment:
I'd like to document the fact that `return 2` block is currently not reached at all. The easiest way is probably to add this:
```
var returnNotReached = new PreProcessTestCheck(x =>
{
    x.Operation.Instance.Kind.Should().NotBe(OperationKind.Literal, "we don't support  multiple branches yet");
    return x.State;
});
```